packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -174,15 +174,13 @@ function parseElement(
         validateElement(ctx, element, parse5Elm);
         validateAttributes(ctx, parsedAttr, element);
         validateProperties(ctx, element);
+    } else {
+        validateTemplate(ctx, parsedAttr, parse5Elm, parse5ElmLocation);
```

Comment:
Side note: It is not obvious here why we should validate the template in the else branch only. You have to open `parseBaseElement` to understand that it always returns element EXCEPT when processing a `<template>`.